src/text/span.rs

```diff
@@ -332,6 +332,37 @@ impl<'a> Span<'a> {
     pub fn to_right_aligned_line(self) -> Line<'a> {
         self.into_right_aligned_line()
     }
+
+    /// Splits the span into two at the given width.
+    ///
+    /// The first span will contain the characters that fit within the given width, and the second
+    /// span will contain the remaining characters.
+    pub(crate) fn split_at_width(&'a self, width: usize) -> (Self, Self) {
+        // There's at least two ways to do this. I chose the second option as it iterates over the
+        // string once at the cost of always allocating two strings regardless of the split point.
+        // 1. iterate the chars to find the split point and then split the spans.
+        // 2. iterate the chars and build the spans at the same time.
+        let mut left = String::new();
+        let mut right = String::new();
```

Comment:
Using iterators or `with_capacity` will improve performance. `push` in a loop will create a bunch of allocs which can be prevented with knowing accurate or best effort width. Iterators also allow for a similar optimization without needing to write that.

Reply:
If there's an obvious fix that can be made which keeps this simple and obvious, and passes the tests please make it. (Especially if you can show there's a perf impact that is fixed with a quick benchmark).

Reply:
https://github.com/ratatui-org/ratatui/pull/1089/commits/d64f785e8705f7da37f9d392746a914086ea7093